flask_resources/context.py

```diff
@@ -68,9 +79,11 @@ def with_resource_requestctx(f):
     """Wrap in resource request context."""
 
     @wraps(f)
-    def inner(*args, **kwargs):
-        with ResourceRequestCtx():
-            return f(*args, **kwargs)
+    def inner(self, *args, **kwargs):
+        resource_request_context = getattr(
```

Comment:
Nice, making it customizable!

Reply:
The way I see it: not having to specify the context would be even better. To achieve that, we reverse the knowledge flow (like we were doing) and have the `BaseView` and decorators inject the appropriate info that are specific to a request.